### ISSUE TITLE
Detect "Mockery" anti-pattern

### ISSUE THREAD

**Issue by h1alexbel** [volodya-lombrozo/jtcop#393]
How about to detect Mockery anti-pattern?

Mockery definition:
Sometimes mocking can be good, and handy. But sometimes developers can lose themselves in their effort to mock out what isn’t being tested. In this case, a unit test contains so many mocks, stubs, and/or fakes that the system under test isn’t even being tested at all, instead data returned from mocks is what is being tested.

Example:

```java
import org.mockito.Mockito;

final class FooTest {

  @Test
  void testsSomething() {
    Frame frame = Mockito.mock(Frame.class);
    Mockito.doReturn(...).when(frame).iterator();
    Table table = Mockito.mock(Table.class);
    Mockito.doReturn(frame).when(table).frame();
    Region region = Mockito.mock(Region.class);
    Mockito.doReturn(table).when(region).table(Mockito.anyString());
 }
}
```

in `pom.xml` we can introduce new parameter `maxNumberOfMocks`:

```xml
<configuration>
  <maxNumberOfMocks>2</maxNumberOfMocks>
</configuration>
```

Thus, using example above, `jtcop` should fail with message like this:

```text
[ERROR]: FooTest.java#testsSomething has too many mocks: 3 (max allowed 2)
```

**Comment by h1alexbel** [comment]
@volodya-lombrozo what do you think?

**Comment by h1alexbel** [comment]
@volodya-lombrozo not yet, but I can take research this topic

**Comment by h1alexbel** [comment]
@volodya-lombrozo assigne me, please

**Comment by volodya-lombrozo** [comment]
@h1alexbel Thank you!

**Comment by 0pdd** [comment]
@h1alexbel 3 puzzles [#396](https://github.com/volodya-lombrozo/jtcop/issues/396), [#397](https://github.com/volodya-lombrozo/jtcop/issues/397), [#398](https://github.com/volodya-lombrozo/jtcop/issues/398) are still not solved.

**Comment by 0pdd** [comment]
@h1alexbel 2 puzzles [#396](https://github.com/volodya-lombrozo/jtcop/issues/396), [#397](https://github.com/volodya-lombrozo/jtcop/issues/397) are still not solved; solved: [#398](https://github.com/volodya-lombrozo/jtcop/issues/398).

**Comment by 0pdd** [comment]
@h1alexbel the puzzle [#397](https://github.com/volodya-lombrozo/jtcop/issues/397) is still not solved; solved: [#396](https://github.com/volodya-lombrozo/jtcop/issues/396), [#398](https://github.com/volodya-lombrozo/jtcop/issues/398).

**Comment by volodya-lombrozo** [comment]
@rultor release, tag is `1.3.0`, title is `1.3.0`.
